src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Unit.Tests/JsonSourceGeneratorTests.cs

```diff
@@ -634,6 +634,39 @@ public class ClassWithPropertiesAndFieldsThatAreReservedKeywords
             CompilationHelper.CheckDiagnosticMessages(DiagnosticSeverity.Error, generatorDiags, Array.Empty<(Location, string)>());
         }
 
+        [Fact]
+        public void EquivalentTupleDeclarations_DoNotConflict()
```

Comment:
Oh is this why we trim type symbol metadata?

Reply:
Correct, compile-time metadata needs to be trimmed regardless of nullability annotation (tuple labels are independent of them).